epick_driver/src/default_driver.cpp

```diff
@@ -138,14 +172,11 @@ void DefaultDriver::activate()
   auto crc = crc_utils::compute_crc(request);
   request.push_back(data_utils::get_msb(crc));
   request.push_back(data_utils::get_lsb(crc));
-  try
-  {
-    serial_->write(request);
-    auto response = serial_->read(kActivateResponseSize);
-  }
-  catch (const serial::IOException& e)
+
+  auto response = send(request, kActivateResponseSize);
+  if (response.empty())
   {
-    RCLCPP_ERROR(kLogger, "Failed to activate the gripper: %s", e.what());
+    RCLCPP_ERROR(kLogger, "Failed to activate the gripper.");
     throw;
```

Comment:
Should this create the exception to be thrown now that it's no longer inside a catch block?

Reply:
I added a custom DriverException.